src/amplitude.js

```diff
@@ -50,38 +51,42 @@ define(function (require) {
 
     // set audio context
     this.audiocontext = p5sound.audiocontext;
-    this.processor = this.audiocontext.createScriptProcessor(this.bufferSize, 2, 1);
+    this._workletNode = new AudioWorkletNode(this.audiocontext, processorNames.amplitudeProcessor, {
+      outputChannelCount: [1],
+      parameterData: { smoothing: smoothing || 0 },
```

Comment:
as a potential improvement, we could check the range here, as in the `smooth` method. But we could also just let the parameterDescriptors handle enforcing (and choosing whether to notify the user) about the range.